File: Surrogate Modelling/Partial-Least-Square-Based Polynomial Chaos Expansion (PLS-PCE)/PLS-PCE_MATLAB/surrogates_MATLAB/eval_quad.m
function xq = eval_quad(nele,L,xGL)

ind = 1:nele;       
l = L/nele;       
XQ = l/2*repmat(xGL',1,length(ind))+l/2*repmat((2*ind-1),length(xGL),1);
xq = XQ(:);

end

File: Surrogate Modelling/Partial-Least-Square-Based Polynomial Chaos Expansion (PLS-PCE)/PLS-PCE_Python/surrogates_Python/eval_quad.py
import numpy as np
from numpy.matlib import repmat

def eval_quad(nele,L,xGL):

    ind = np.arange(1,nele+1)      
    l = L/nele       
    XQ = l/2 * repmat(xGL.T, 1, len(ind)) + l/2 * repmat( (2*ind-1), len(xGL),1)
    xq = np.reshape(XQ.T,(300,1))

    return xq